.github/workflows/bazeltest.yml

```diff
@@ -22,12 +22,14 @@ jobs:
 
     steps:
       - uses: actions/checkout@v2
+      - uses: bazel-contrib/setup-bazel@0.8.5
```

Comment:
It's not immediately clear to me how (or if) this specifies the `bazel` version to install. IIRC, this is important for qsim as it needs to align with the bazel version used in [TFQ](https://github.com/tensorflow/quantum) for compatibility reasons.

Can we lock the bazel version to `5.3.0` here?

Reply:
The bazel version is locked by `.bazelversion` in the root of the repo, and that's already set to `5.3.0`.

`setup-bazel` will install bazelisk 1.x as `bazel` and then any time `bazel` is called, bazelisk will download and install the bazel version defined by `.bazelversion` if the file exists (which it does :upside_down_face:)

More info on how bazelisk determines which version of bazel to run can be found here: https://github.com/bazelbuild/bazelisk?tab=readme-ov-file#how-does-bazelisk-know-which-bazel-version-to-run